static/app/components/events/eventCustomPerformanceMetrics.tsx

```diff
@@ -2,24 +2,29 @@ import styled from '@emotion/styled';
 import {Location} from 'history';
 
 import {SectionHeading} from 'sentry/components/charts/styles';
+import DropdownMenuControl from 'sentry/components/dropdownMenuControl';
 import FeatureBadge from 'sentry/components/featureBadge';
 import {Panel} from 'sentry/components/panels';
+import {IconEllipsis} from 'sentry/icons';
 import {t} from 'sentry/locale';
 import space from 'sentry/styles/space';
 import {Organization} from 'sentry/types';
 import {Event} from 'sentry/types/event';
+import EventView from 'sentry/utils/discover/eventView';
 import {
   DURATION_UNITS,
   FIELD_FORMATTERS,
   PERCENTAGE_UNITS,
   SIZE_UNITS,
 } from 'sentry/utils/discover/fieldRenderers';
 import {isCustomMeasurement} from 'sentry/views/dashboardsV2/utils';
+import {transactionSummaryRouteWithQuery} from 'sentry/views/performance/transactionSummary/utils';
 
 type Props = {
   event: Event;
   location: Location;
   organization: Organization;
+  fromPerformance?: boolean;
```

Comment:
is this component going to be used in places other than discover and performance? issues perhaps? is it worth future-proofing this to accept an enum source: Performance or Discover so we don't have to add more boolean fields?

Reply:
Yeah good point, I was thinking about refactoring this since it didnt really seem extensible to me. I've updated it to use an enum for now (either `performance` or `discover`) and we can extend this enum as we add other sources like issues.